### PR TITLE
Added pluggable Flag models

### PR DESCRIPTION
**Pull request by koliber**

### CODE REVIEW

File: docs/types/flag.rst [hunk 1]
```diff
@@ -84,6 +84,98 @@ are in the group *or* if they are in the 12%.
...
     probably differ slightly from the Percentage value.
+    
+    
+.. _types-flag-custom-model:
+
+Custom Flag Models
+======================
+
+For many cases, the default Flag model provides all the necessary functionality. It allows
+flagging individual ``User``s and ``Group``s. If you would like to flags be applied to
```
Comment (clintonb) [inline docs/types/flag.rst]: If you would like flags to...

File: docs/types/flag.rst [hunk 2]
```diff
@@ -84,6 +84,98 @@ are in the group *or* if they are in the 12%.
...
+different things, such as companies a User belongs to, you can use a custom flag model for that.
```
Comment (clintonb) [inline docs/types/flag.rst]: Remove "for that" at the end of the sentence.

**Review by clintonb — COMMENTED** [review]
This change has come from nowhere. It would have been nice to have some discussion prior to the PR. Nevertheless, the contribution is appreciated.

Did you consider other implementations of swappable models? Oscar, for example, has an implementation that allows sub-apps to be "forked" at anytime during the life of the project, not only at the beginning. See http://django-oscar.readthedocs.io/en/latest/topics/fork_app.html.

**Comment by koliber** [comment]
Apologies for not adding details to this MR. 

I had an PR for organization support on a flag about 2 years ago, but it was primitive. @jsocol recommended a swappable model approach. Last year at DiangoCon Europe I got a suggestion from Russel Keith Magee to use the same method that Django uses. I made a PR and it lingered without comment. I created this MR again using the Django-like approach now as I would really like to see this support in Waffle!

I am not familiar with Oscar's swappable model. I will read up on that.

**Comment by jsocol** [comment]
I'm good with this approach, it follows #181, and I think, since Flags are the only models (at the moment) with relationships or multiple rules, it makes sense to limit the ability to override the model to flags.

**Comment by hugobessa** [comment]
@jsocol and @clintonb, is this PR ready to be merged?

**Comment by clintonb** [comment]
I'll defer to @jsocol since he has the most context/history with this change. Also, I'm a bit busy this weekend, and won't have time to dedicate to reviewing/releasing this.

**Comment by koliber** [comment]
Is there anything I can help with to get this PR reviewed and merged?

**Comment by twymer** [comment]
This PR solves the exact use case I was hoping to use Django Waffle for. Want to be able to turn on/off features based on the company that a user belongs to. Would love to see this make it into the project!

**Comment by koliber** [comment]
@jsocol are you interested in including this change in Django Waffle? If not, I will close this merge request.

**Comment by intiocean** [comment]
+1 Basically re-iterating what @twymer said - I'd love to use this. I was planning to implement my own feature flags to enable features based on the company a user belongs to but if this was merged I'd use it! :)

**Comment by cspanring** [comment]
Came here investigating a solution for the use case @twymer described. Would be great to see this change landing in django waffle!

**Comment by koliber** [comment]
A lot of SaaS companies have an organization as a subscriber. Individual Django `User` instances belong to the organization. Very often, features are flipped on the organization level instead of on the user level. 

Until this PR (or another one implementing similar functionality) gets merged in, such projects can not use `django-waffle` for feature management. A whole segment of potential adopters is unable to use this fine piece of software because of this missing piece of functionality.

**Review by clintonb — APPROVED** [review]
Thank you for the contribution. I can approve, and merge, this change; however, I am unable to create a new release without @jsocol approving that PR.

**Comment by ndibari-sp** [comment]
Hi @clintonb @jsocol has there been any discussion on cutting a new version of django-waffle to get this feature into the next release?

**Comment by clintonb** [comment]
@ndibari-sp see https://github.com/django-waffle/django-waffle/releases/tag/v0.15.0

**Pull request merged**